src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -454,6 +457,66 @@ public AgentSettings LoadSettings()
             }
         }
 
+#if OS_WINDOWS
+        private void ConfigureAutoLogonIfNeeded(CommandSettings command)
+        {
+            Trace.Info(nameof(ConfigureAutoLogonIfNeeded));
+
+            if(!command.GetEnableAutoLogon())
+            {
+                return;
+            }
+
+            Trace.Info("Agent is going to run as process setting up the 'AutoLogon' capability for the agent.");
+            try
+            {                
+                var autoLogonConfigManager = HostContext.GetService<IAutoLogonConfigurationManager>();
+                autoLogonConfigManager.Configure(command);
+
+                if(autoLogonConfigManager.RestartNeeded())
+                {
+                    Trace.Info("AutoLogon is configured for a different user than the current user. Machine needs a restart.");
+                    _term.WriteLine(StringUtil.Loc("RestartMessage"));
+                    var shallRestart = command.GetRestartNow();
+                    if(shallRestart)
+                    {
+                        var whichUtil = HostContext.GetService<IWhichUtil>();
+                        var shutdownExePath = whichUtil.Which("shutdown.exe");
+
+                        Trace.Info("Restarting the machine in 5 seconds");
+                        _term.WriteLine(StringUtil.Loc("RestartIn5SecMessage"));
+                        string msg = StringUtil.Loc("ShutdownMessage");
+                        //we are not using ProcessInvoker here as today it is not designed for 'fire and forget' pattern
+                        //ExecuteAsync API of ProcessInvoker waits for the process to exit
+                        Process.Start(shutdownExePath, $"-r -t 5 -c {msg}");
+                    }
+                    else
+                    {
+                        Trace.Info("No restart happened. As the interactive session is configured for a different user agent will not be launched");
+                    }
+                    return;
+                }
+            }
+            catch(Exception ex)
+            {
+                Trace.Error(ex);
+                _term.WriteLine(StringUtil.Loc("AutoLogonConfigurationFailureMessage"));
+            }
+        }
+
+        private void UnConfigureAutoLogonIfNeeded()
```

Comment:
Can you delete this method and inline the contents instead so the unconfig flow follows the same readability pattern as the rest of the code. The readability of config/unconfig can quickly get out of hand. We put a lot of effort into cleaning this up in the 2x version of the agent. Please fit the additions you are making here into the same established pattern in this file.